kms_log/main.tf

```diff
@@ -558,4 +593,469 @@ resource "aws_cloudwatch_metric_alarm" "dead_letter" {
+            "logs:CreateLogGroup"
+        ]
+
+        resources = [
+            "arn:aws:logs:${var.region}:${data.aws_caller_identity.current.account_id}:*",
+        ]
+    }
+    statement {
+        sid = "PutLogEvents"
+        effect = "Allow"
+        actions = [
+            "logs:CreateLogStream",
+            "logs:PutLogEvents"
+        ]
+
+        resources = [
+            "arn:aws:logs:${var.region}:${data.aws_caller_identity.current.account_id}:log-group:/aws/lambda/${aws_lambda_function.cloudwatch_processor.id}:*"
+        ]
+    }
+}
+
+data "aws_iam_policy_document" "cwprocessor_dynamodb" {
+    statement {
+        sid = "DynamoDb"
+        effect = "Allow"
+        actions = [
+            "dynamodb:PutItem",
+            "dynamodb:GetItem"
+        ]
+
+        resources = [
+            "${aws_dynamodb_table.kms_events.arn}"
+        ]
+    }
+}
+
+data "aws_iam_policy_document" "cwprocessor_kms" {
+    statement {
+        sid = "KMS"
+        effect = "Allow"
+        actions = [
+            "kms:Decrypt",
+            "kms:GenerateDataKey"
+        ]
+
+        resources = [
+            "${aws_kms_key.kms_logging.arn}"
+        ]
+    }
+}
+
+data "aws_iam_policy_document" "cwprocessor_sns" {
+    statement {
+        sid = "SNS"
+        effect = "Allow"
+        actions = [
+            "sns:Publish"
+        ]
+
+        resources = [
+            "${aws_sns_topic.kms_logging_events.arn}"
+        ]
+    }
+}
+
+data "aws_iam_policy_document" "cwprocessor_kinesis" {
+    statement {
+        sid = "Kinesis"
+        effect = "Allow"
+        actions = [
+            "kinesis:GetShardIterator",
+            "kinesis:GetRecords",
+            "kinesis:DescribeStream"
+        ]
+
+        resources = [
+            "${aws_kinesis_stream.datastream.arn}"
+        ]
+    }
+}
+
+resource "aws_iam_role_policy" "cwprocessor_cloudwatch" {
+    name = "CloudWatch"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_cloudwatch.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_dynamodb" {
+    name = "DynamoDb"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_dynamodb.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_kms" {
+    name = "KMS"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_kms.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_sns" {
+    name = "KMS"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_sns.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_kinesis" {
+    name = "Kinesis"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_kinesis.json}"
+}
+
+# lambda for creating cloudwatch metrics and events
+resource "aws_lambda_function" "event_processor" {
```

Comment:
This looks fine but it probably needs to be commented out because it isn't in identity-lambda-functions yet.

Reply:
Alternately we could create a dummy lambda in i-l-f that does nothing for now.

Reply:
Nevermind. It's fine like this. I'm an idiot.